protocol/metrics/metrics_consumer_manager.go

```diff
@@ -13,6 +13,20 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+type WsDisconnectReasonEnum int
+
+const (
```

Comment:
instead of the iota and the map it is better to just have a method with switch case and each one would write a different reason 

```golang
reason := ""
switch disconnectionReason {
case WS_DISCONNECTION_REASON_CONSUMER: 
    reason = "consumer-disconnect"
// other cases ... 
}
 ```